programs/drift/src/state/perp_market.rs

```diff
@@ -441,6 +441,23 @@ impl PerpMarket {
 
         Ok(depth)
     }
+
+    pub fn update_market_with_counterparty(
+        &mut self,
+        delta: &PositionDelta,
+        new_settled_base_asset_amount: i64,
+    ) -> DriftResult {
+        if delta.remainder_base_asset_amount.is_some() {
```

Comment:
why check if this is some and not `new_settled_base_asset_amount` is not equal to zero?